addons/sale/models/sale_order.py

```diff
@@ -768,8 +769,11 @@ def action_confirm(self):
                 'It is not allowed to confirm an order in the following states: %s'
             ) % (', '.join(self._get_forbidden_state_confirm())))
 
+        mapped_subscriptions = defaultdict(lambda: self.env['sale.order'])
         for order in self.filtered(lambda order: order.partner_id not in order.message_partner_ids):
-            order.message_subscribe([order.partner_id.id])
+            mapped_subscriptions[order.partner_id.id] += order
+        for partner_id, orders in mapped_subscriptions.items():
+            orders.message_subscribe([partner_id])
```

Comment:
We can store records directly and use `.ids`

```suggestion
            mapped_subscriptions[order.partner_id] += order
        for partner_id, orders in mapped_subscriptions.items():
            orders.message_subscribe(partner_id.ids)
```